### PR TITLE
Soften errors for invalid @objc attributes added by access notes

### PR DESCRIPTION
**Pull request by beccadax**
This PR modifies the existing `@objc` validation code to reduce its errors to remarks when the `@objc` attribute was added by an access note. It builds on previous work to add an `InFlightDiagnostic::limitBehavior()` method and use it during `@objc` checking so that it doesn't add a lot of new conditional paths or duplicate errors.

I have also adapted nearly all of the test cases in test/attr/attr_objc.swift for access notes without actually duplicating the code by annotating the tests with comments and generating test files from it using a small tool. These test changes account for most of the lines touched by this PR. This testing approach also shook out half a dozen bugs in access notes, which have all been fixed by the end of this PR.

Fixes rdar://71875410.

-----

Review instructions:

* ~~"Add DiagnosticStateRAII" and "Use behavior limits to control @objc errors" are being reviewed separately in #36474; you can ignore them until they're rebased away.~~
* "Add -access-notes-path to swift-ide-test" just adds a command-line flag to a test tool that we'll use later.
* "Regularize @objc use in attr/attr_objc.swift" is literally just adding newlines after `@objc` attributes which don't already have them. Skim it to make sure there are absolutely no functional changes to the test.
* "Diagnose access note @objc failures using remarks" is the biggie. Read the compiler changes, the new Python test-helper tool, and the changes at the top of attr/attr_objc.swift. Skim over the changes in the rest of attr/attr_objc.swift to make sure you don't see anything that looks like it would affect the meaning of an existing test.
* The remaining commits all fix tests that were broken in the new access-note-based RUN lines of attr/attr_objc.swift. Make sure these changes all look reasonable.

### CODE REVIEW

**Comment by beccadax** [comment]
@swift-ci please test

**Comment by beccadax** [comment]
@swift-ci please test

**Comment by beccadax** [comment]
@swift-ci please smoke test

File: test/attr/Inputs/access-note-gen.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+# Offsets
+#
+
+"""Matches an @±N offset."""
+offset_re_fragment = r'[ \t]*(?:@([+-]\d+))?[ \t]*'
+
+
+def offsetify(*offsets):
+    """Sum line offsets matched by offset_fragment_re and convert them to strings
+       like @+3 or @-2."""
```
Comment (xymus) [inline test/attr/Inputs/access-note-gen.py:61]: Should this reference `offsets` or `offset_re_fragment` instead of `offset_fragment_re`?

**Review by xymus — APPROVED** [review]
This looks good!

File: test/attr/Inputs/access-note-gen.py (hunk 1 above)
Comment (beccadax) [reply test/attr/Inputs/access-note-gen.py:61]: You're right, it should say `offset_re_fragment`.

**Comment by beccadax** [comment]
In addition to the comment fix pointed out by @xymus, I've also fixed another incorrect comment and made `setAddedByAccessNote()` get called on all access-note-added attributes instead of just ObjCAttrs. Very minor changes, so I'll just do this:

@swift-ci please smoke test and merge

**Comment by beccadax** [comment]
@swift-ci please smoke test linux platform

**Pull request merged**